staging/cse/windows/provisioningscripts/cleanupnetwork.ps1

```diff
@@ -29,16 +39,38 @@ if ($hnsNetwork) {
     # https://github.com/delulu/kubernetes/blob/524de768bb64b7adff76792ca3bf0f0ece1e849f/pkg/proxy/winkernel/proxier.go#L532
     Get-HnsPolicyList | Remove-HnsPolicyList
 
-    Write-Host "Cleaning up old HNS network found"
-    Remove-HnsNetwork $hnsNetwork
-    Start-Sleep 10
+    if ($global:IsSkipCleanupNetwork) {
+        Write-Log "Remove HNS Endpoint before removing HNS Network."
+        Get-HnsEndpoint | Remove-HnsEndpoint
+        Remove-HnsNetwork $hnsNetwork
+
+        Write-Log  "Count actively reserved port pools to log."
+        $countPortPools = 1
+        # Guarantee (10s <= wait time <= 60s) between HNS removal vs. creation for safety.
+        # Usually, it takes 10s to sleep like before, but it may take longer in some cases (e.g., COSMIC).
+        $maxIteration=6
+        For ($i=1; ($i -le $maxIteration -and $countPortPools -ne 0); $i++) {
+            $countPortPools = (Invoke-HnsRequest -Method "GET" -Type "portpools" | Select-Object -ExpandProperty PortPoolAllocations | Where-Object -FilterScript {$_.RequiresHostPortReservation -eq "True" -and $_.PortRanges} | Select-Object -ExpandProperty PortRanges).Count
```

Comment:
do we need to use try/catch to avoid unexpected error in this block?

Reply:
Agree. Updated code and test:
![image](https://github.com/Azure/AgentBaker/assets/62196586/90dc4de1-5207-40f7-bbe1-a981702cc2e3)